build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"io/fs"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type AptRepoTool struct {
+	config       *Config
+	aptly        *Aptly
+	s3Manager    *s3manager
+	supportedOSs *map[string][]string
+}
+
+// Instantiates a new apt repo tool instance and performs any required setup/config.
+func NewAptRepoTool(config *Config, supportedOSs *map[string][]string) (*AptRepoTool, error) {
```

Comment:
Why not make `supportedOSs` a part of `config` too?

Reply:
I thought about that... but I figured building the map in Go would be significantly simpler compared to passing it in via CLI and parsing it. Or are you asking why I don't move the supported OS struct from `main.go` to `config.go` and adding it to the struct?